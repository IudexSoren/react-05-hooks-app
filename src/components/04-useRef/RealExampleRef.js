import { useState } from 'react';
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks';

import '../02-useEffect/effects.css';

const RealExampleRef = () => {

  const [show, setShow] = useState(false);

  return (
    <div>
      <h1>RealExampleRef</h1>
      <hr/>
      { show && <MultipleCustomHooks /> }
      <button
        className="btn btn-outline-dark mt-3"
        onClick={ () => { setShow(!show) } }
      >{ (show) ? 'Hide' : 'Show' } quote</button>
    </div>
  )
}

export default RealExampleRef;
