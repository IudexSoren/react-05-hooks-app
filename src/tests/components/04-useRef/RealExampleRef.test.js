import { shallow } from 'enzyme';
import RealExampleRef from '../../../components/04-useRef/RealExampleRef';

describe('Pruebas de RealExampleRef', () => {

  const wrapper = shallow(<RealExampleRef />);

  test('Debe renderizar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBeFalsy()
  });

  test('Debe mostrar el componente <MultipleCustomHooks />', () => {
    wrapper.find('button').simulate('click');

    expect(wrapper.find('MultipleCustomHooks').exists()).toBeTruthy();
  });

});
