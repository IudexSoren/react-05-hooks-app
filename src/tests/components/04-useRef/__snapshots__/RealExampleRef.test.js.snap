// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de RealExampleRef Debe renderizar el componente correctamente 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-outline-dark mt-3"
    onClick={[Function]}
  >
    Show
     quote
  </button>
</div>
`;
